internal/reconcilers/reconcilers.go

```diff
@@ -39,24 +39,14 @@ type Reconciler struct {
 	fallbackTokenClient *gogithub.Client
 }
 
-// ReconcilerOption is a function that modifies a reconciler
-type ReconcilerOption func(*Reconciler)
-
-// WithProviderMetrics sets the provider metrics for the reconciler
-func WithProviderMetrics(mt providertelemetry.ProviderMetrics) ReconcilerOption {
-	return func(r *Reconciler) {
-		r.provMt = mt
-	}
-}
-
 // NewReconciler creates a new reconciler object
 func NewReconciler(
 	store db.Store,
 	evt events.Publisher,
 	authCfg *serverconfig.AuthConfig,
 	provCfg *serverconfig.ProviderConfig,
 	restClientCache ratecache.RestClientCache,
-	opts ...ReconcilerOption,
+	providerMetrics providertelemetry.ProviderMetrics,
```

Comment:
In my next PR, ProviderBuilder will be pass as a dependency, and thus simplify the wireup further.